### PR TITLE
Creating notifications

### PR DESCRIPTION
**Pull request by Favor-star**
## Purpose

The purpose of this pull request is to create the notifications related to the product creations, product update and their deletions as well.

## Changes Made

Initially, they were a creation of the following fiiles:
```
src/database/migrations/******-create-notification.js
src/databse/models/notification.ts
src/controllers/notificationControllers.ts
src/controllers/notificationRoutes.ts
```
These files were also changed, as they were some contents that had to be added. However, nothing deleted compared to what was initially there.
```
src/helpers/send-email.ts
src/controllers/productControllers.ts

```
 we are planning to include hooks in the product controllers; so that when the 

## Testing Instructions

No tests made yet!

## Related Issues

Products is not yet merged, so I was unable to use it into my branch!

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:

- [x] Code follows the project's coding standards
- [x] Changes are covered by tests
- [x] Documentation is updated (if applicable)
- [x] All tests pass

### CODE REVIEW

File: src/controllers/notificationsController.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import { NotificationAttributes, NotificationCreationAttributes } from '../database/models/notification';
+Role.afterCreate((role, options) => {
+  try {
+    console.log('Role was created successfully', role);
+    // Any additional logic you want to execute after a Role is created
+  } catch (error) {
+    console.error('Error occurred after creating Role:', error);
+  }
+});
+
```
Comment (niyontwali) [inline src/controllers/notificationsController.ts]: Please review this once again and remove the consoles

**Review by hozayves — APPROVED** [review]

File: src/controllers/notificationsController.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { Request, Response } from 'express';
+import Notification from '../database/models/notification';
+import { sendInternalErrorResponse } from '../validations';
+import logger from '../logs/config';
+import Role from '../database/models/role';
+import { NotificationAttributes, NotificationCreationAttributes } from '../database/models/notification';
+Role.afterCreate((role, options) => {
+  try {
+    console.log('Role was created successfully', role);
```
Comment (JeanIrad) [inline src/controllers/notificationsController.ts]: consider removing this log function

Comment (patrickhag) [inline src/controllers/notificationsController.ts]: why bother using using console yet we have winston logger?

File: src/controllers/notificationsController.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    // Any additional logic you want to execute after a Role is created
+  } catch (error) {
+    console.error('Error occurred after creating Role:', error);
```
Comment (patrickhag) [inline src/controllers/notificationsController.ts]: and even here

**Review by patrickhag — COMMENTED** [review]
your PR looks cool and ready but consider changing this ASAP, it might take you 3 minutes.

File: src/controllers/notificationsController.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  } catch (error) {
+    logger.error(error);
+    sendInternalErrorResponse(res, error);
+    return;
+  }
+};
+
+export const markNotificationAsRead = async (req: Request, res: Response): Promise<void> => {
+  const { id } = req.params;
+  const { isRead } = req.body;
```
Comment (amin-leon) [inline src/controllers/notificationsController.ts:28]: why are passing isRead in body? only notification id is enough to change isRead to true once user clicks on notification

**Review by amin-leon — APPROVED** [review]

File: src/controllers/notificationsController.ts (hunk 4 above)
Comment (Favor-star) [reply src/controllers/notificationsController.ts:28]: but I think this will help the user to track the current status of the notification!

**Review by hozayves — APPROVED** [review]

**Review by P-Rwirangira — APPROVED** [review]

**Review by niyontwali — APPROVED** [review]

**Pull request merged**